<h2>Setting up our data</h2>

Get our <b>clean CSV</b> file

In [5]:
from pathlib import Path
import pandas as pd
def get_clean_csv():
    clean_csv_path = Path("datasets/medical_insurance/clean_insurance_data.csv")
    return pd.read_csv(clean_csv_path)
    
insurance = get_clean_csv()

<h2>BMI mean confidence interval </h2>

In [6]:
from scipy import stats

We calculate the range of our <b>95% confidence interval</b> on <b>bmi</b> true polpulation <b>mean</b>

In [7]:
bmi_data = insurance['bmi']
bmi_mean_point_estimate = bmi_data.mean()
normalize_error = stats.sem(bmi_data) # s / sqrt(n) 
n = len(bmi_data)
degrees_of_freedom = n - 1
lower_interval, upper_interval = stats.t.interval(confidence = 0.95, loc = bmi_mean_point_estimate, df = degrees_of_freedom, scale = normalize_error)
print(f"Mean Point Estimate: {bmi_mean_point_estimate:.4f}")
print(f"n: {n}")
print(f"s: {bmi_data.std():.4f}")
print(f"Confidence Range 95%: ({lower_interval:.4f} , {upper_interval:.4f})")

Mean Point Estimate: 30.7928
n: 1338
s: 5.9522
Confidence Range 95%: (30.4735 , 31.1120)


<ol>Here is an explaination about our assumptions
    <li> We assume that the polpulation mean bmi is unknown, therefore we calculate the estimate interval using sample mean <b>s</b> and <b>t-test</b>
    <li>We assume the dataset is a randomly and independently representation of the whole polpulation</li>
    <li> Even though t-test require a sampling distribution of mean approximate to normal, which in our case there are sone outliers. But the number of observation is large (1338) and the outliers themselves are not that big (proved ealier with a tiny right tail skewed histogram) and there are only 9 of them. Therefore, by using the <b>Central Limit Theorem, this assumption is satisfied</li>
</li>
</ol>

<b>Conclusion:</b> We find the 95% interval range is between 30.4735 and 31.1120. Therefore, we are 95% confident that the average patient bmi lies in the range between 30.4735 and 31.1120. Which mean that if we take a 100 random samples batches of bmi from patients, then construct the distribution of each batch and apply the found interval for each. Then 95 times out of 100, we would successfully capture the true polpulation mean.

<h2>Hypothesis</h2>

<h3>Research question</h3>

Ho Chi Minh city surveyed that around 40% of the citizens are overweight. Note that according to WHO, overweight people are considered to have their <b>bmi larger than 25</b>. So we would want to verify <b>if the percentage of overweight patient in our dataset is more than Ho Chi Minh polpulation</b> to see if our patient is generally more unhealthier than Ho Chi Minh polpulation

<h3>Hypothesis</h3>

<ul>
    <li>p0 = 40%</li>
    <li>Null hypothesis (H0): p <= 40% (Propotion of our patient having bmi lower or equal to the propotion Ho Chi Minh city polpulation)</li>
    <li>Alternate hypothesis (H1): p > 40% (Propotion of our patient having bmi higher than the propotion of Ho Chi Minh city polpulation)</li>
    <li>Significance level $\alpha$ = 0.05</li>
</ul>

First looking at the dataset of the patient we found

In [8]:
import math
p0 = 0.4
overweight = insurance[insurance['bmi'] > 25]
n = len(insurance)
number_of_overweight = len(overweight)
overweight_ratio = number_of_overweight / len(insurance)
z_standard = (overweight_ratio - p0) / math.sqrt((p0 * (1 - p0)) / n) # f - p0 / sqrt((p0 * (1 - p0)) / n)

#looking up z, right tail
p = 1 - stats.norm.cdf(z_standard)

print(f"Number of overweight patient: {number_of_overweight}")
print(f"n (total patient): {n}")
print(f"Overweight ratio: {overweight_ratio:.4f}")
print(f"Z standard: {z_standard:.4f}")
print(f"p value: {p}")

if p < 0.05:
    print(f"p ({p}) < alpha (0.05): We reject H0")
else:
    print(f"p ({p}) >= alpha (0.05): Insuficient evidence to reject H0")

Number of overweight patient: 1111
n (total patient): 1338
Overweight ratio: 0.8303
Z standard: 32.1320
p value: 0.0
p (0.0) < alpha (0.05): We reject H0


<b>Conclusion:</b> Since our p value is less the the significance level, we reject the hypthesis H0. Therefore we have sufficient evidence to conclude that the propotion of overweight patient in the hospital is more than that 40% of Ho Chi Minh city overweight polpulation